yarn-project/foundation/src/serialize/types.ts

```diff
@@ -0,0 +1,61 @@
+/**
+ * Represents a fixed-length array.
+ */
+export type TupleOf<T, N extends number> = N extends N ? (number extends N ? T[] : _TupleOf<T, N, []>) : never;
+/**
+ * Recursive type helper for constructing a fixed-length tuple of a given type.
+ * This is utilized internally by TupleOf to create the final fixed-length tuple.
+ */
+type _TupleOf<T, N extends number, R extends unknown[]> = R['length'] extends N ? R : _TupleOf<T, N, [T, ...R]>;
+
+/**
+ * Check an array size, and cast it to a tuple.
+ * @param array - The array.
+ * @param n - The size.
+ * @returns The case tuple, or throws Error.
+ */
+export function toTupleOf<T, N extends number>(array: T[], n: N): TupleOf<T, N> {
+  if (array.length !== n) {
+    throw new Error("Wrong 'fixed array' size");
+  }
+  return array as any;
```

Comment:
Why `as any` instead of `as TupleOf<T, N>`?

Reply:
This is a really subtle call, it's already a 'trust me bro' cast and I just decided to put TupleOf<T,N> in the return type instead